includes/base/controls-stack.php

```diff
@@ -1877,25 +1916,54 @@ protected function _content_template() {}
 	protected function init_controls() {
 		Plugin::$instance->controls_manager->open_stack( $this );
 
-		$this->_register_controls();
+		// TODO: This is for backwards compatibility starting from 2.9.0
+		// This `if` statement should be removed when the method is removed
+		if ( method_exists( $this, '_register_controls' ) ) {
+			$this->_register_controls();
+		} else {
+			$this->_register_controls();
```

Comment:
```suggestion
			$this->register_controls();
```
Move it to version 3.0.0